<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>900</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1100</width>
    <height>900</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1100</width>
    <height>900</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Sunnydale School District - Student Portal</string>
  </property>
  <property name="whatsThis">
   <string/>
  </property>
  <property name="styleSheet">
   <string notr="true">QMenuBar {
    background: none;
    height: 0px;
}

/* === GENERAL APP STYLE === */
QMainWindow {
    font-family: 'Segoe UI', sans-serif;
    font-size: 14px;
    color: #333;
}

 QWidget {
    background-color: rgb(255, 252, 228);
    font-family: 'Segoe UI', sans-serif;
    font-size: 14px;
    color: #333;
}
/* === PUSH BUTTONS === */
QPushButton {
    background-color: rgb(5, 60, 171);
    color: white;
    border: none;
    border-radius: 6px;
    padding: 6px 12px;
}

QPushButton:hover {
    background-color: rgb(187, 228, 255);
}

QPushButton:pressed {
    background-color: #2e5aa0;
}

/* === COMBOBOX, INPUTS, TEXTAREA === */
QComboBox, QLineEdit, QTextEdit, QDateEdit, QTimeEdit {
    background-color: #ffffff;
    border: 1px solid #ccc;
    border-radius: 4px;
    padding: 4px 6px;
}

QComboBox::drop-down {
    border: none;
    background-color: #ffffff;
}

QComboBox QAbstractItemView {
    background-color: #ffffff;
    border: 1px solid #ccc;
    selection-background-color: #e6f0ff;
    selection-color: #000;
}


/* === TABLES === */
QTableWidget, QTableView {
    background-color: #ffffff;
    alternate-background-color: #f0f0f0;
    gridline-color: #dcdcdc;
    border: 1px solid #ccc;
    border-radius: 4px;
}

QHeaderView::section {
    background-color: #e6e9ed;
    padding: 4px;
    border: none;
    font-weight: bold;
}

/* === LIST WIDGETS === */
QListWidget {
    background-color: #ffffff;
    border: 1px solid #ccc;
    border-radius: 4px;
}

/* === CALENDAR === */
QCalendarWidget QToolButton {
    color: #4a90e2;
}

QCalendarWidget QAbstractItemView {
    selection-background-color: rgb(5, 60, 171);
    selection-color: white;
}

/* === GROUP BOXES === */
QGroupBox {
    border: 1px solid #aaa;
    border-radius: 6px;
    margin-top: 6px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 4px;
    background-color: transparent;
    color: #4a90e2;
    font-weight: bold;
}

/* === TABS === */
QTabWidget::pane {
    border: 1px solid #ccc;
    border-radius: 4px;
	background-color: rgb(187, 228, 255);
}

QTabBar::tab {
    background: #e6e9ed;
    border: 1px solid #ccc;
    border-bottom: none;
    padding: 6px 12px;
    border-top-left-radius: 4px;
    border-top-right-radius: 4px;
}

QTabBar::tab:selected {
    background: #ffffff;
    color: rgb(5, 60, 171);
    padding: 6px 12px;
   	font-weight: bold;
}

/* === TOOLTIP === */
QToolTip {
    background-color: #333;
    color: #fff;
    padding: 4px;
    border-radius: 4px;
}

QScrollBar:vertical, QScrollBar:horizontal {
    background: white;
    border: 1px solid #ccc;
    border-radius: 6px;
}

QCheckBox {
    background-color: white;
}

QMessageBox {
    background-color: white;
    font-size: 15px;
}

QMessageBox QLabel {
    background-color: transparent;
}


</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_7">
    <property name="leftMargin">
     <number>20</number>
    </property>
    <property name="topMargin">
     <number>20</number>
    </property>
    <property name="rightMargin">
     <number>20</number>
    </property>
    <property name="bottomMargin">
     <number>20</number>
    </property>
    <item>
     <widget class="QFrame" name="frame">
      <property name="whatsThis">
       <string/>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <layout class="QHBoxLayout" name="sunnydale_header">
         <property name="sizeConstraint">
          <enum>QLayout::SetMinimumSize</enum>
         </property>
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="sunnydale_header_png">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
             <horstretch>1</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string/>
           </property>
           <property name="pixmap">
            <pixmap>:/sunnydale_header.png/sunnydale_header.png</pixmap>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="logout_button">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <family>Segoe UI,sans-serif</family>
             <pointsize>15</pointsize>
            </font>
           </property>
           <property name="styleSheet">
            <string notr="true">
   	font-size: 15pt;
</string>
           </property>
           <property name="text">
            <string>Logout</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <widget class="QWidget" name="Dashboard">
       <attribute name="title">
        <string>  Dashboard  </string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QFrame" name="dashboard_frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <item>
            <widget class="QLabel" name="labelStudentName_2">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="styleSheet">
              <string notr="true">#labelStudentName_2 {
	qproperty-alignment: 'AlignCenter';
   	font-size: 20pt;
   	font-weight: bold;
}</string>
             </property>
             <property name="text">
              <string>Welcome, [DYNAMIC: Student Name]</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="student_id_label">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="maximumSize">
              <size>
               <width>1200</width>
               <height>1000</height>
              </size>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTableWidget" name="student_course_table">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>1</verstretch>
              </sizepolicy>
             </property>
             <property name="verticalScrollBarPolicy">
              <enum>Qt::ScrollBarAlwaysOff</enum>
             </property>
             <property name="horizontalScrollBarPolicy">
              <enum>Qt::ScrollBarAlwaysOff</enum>
             </property>
             <property name="sizeAdjustPolicy">
              <enum>QAbstractScrollArea::AdjustToContents</enum>
             </property>
             <column>
              <property name="text">
               <string>Course ID</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Course Name</string>
              </property>
             </column>
             <column>
              <property name="text">
               <string>Time</string>
              </property>
             </column>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QCalendarWidget" name="calendarWidget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumDate">
           <date>
            <year>2025</year>
            <month>3</month>
            <day>1</day>
           </date>
          </property>
          <property name="maximumDate">
           <date>
            <year>2025</year>
            <month>5</month>
            <day>31</day>
           </date>
          </property>
          <property name="gridVisible">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="Grades">
       <attribute name="title">
        <string>  Grades  </string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QFrame" name="grade_frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_2">
           <item>
            <widget class="QFrame" name="overall_grade_frame">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_11">
              <item>
               <widget class="QLabel" name="overall_grade_label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="styleSheet">
                 <string notr="true">#overall_grade_label {
	qproperty-alignment: 'AlignCenter';
   	font-size: 18pt;
   	font-weight: bold;
}</string>
                </property>
                <property name="text">
                 <string>Course Grades</string>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QTableWidget" name="tableWidgetStudentGrades">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="verticalScrollBarPolicy">
                 <enum>Qt::ScrollBarAlwaysOff</enum>
                </property>
                <property name="horizontalScrollBarPolicy">
                 <enum>Qt::ScrollBarAlwaysOff</enum>
                </property>
                <property name="sizeAdjustPolicy">
                 <enum>QAbstractScrollArea::AdjustToContents</enum>
                </property>
                <column>
                 <property name="text">
                  <string>Course Name</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Weighted Average</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Letter Grade</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="assignment_grade_frame">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_10">
              <item>
               <widget class="QLabel" name="course_assignment_label">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="styleSheet">
                 <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 18pt;
   	font-weight: bold;
</string>
                </property>
                <property name="text">
                 <string>Assignment Grades</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QComboBox" name="course_assignment_selector"/>
              </item>
              <item>
               <widget class="QTableWidget" name="course_assignment_table">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="verticalScrollBarPolicy">
                 <enum>Qt::ScrollBarAlwaysOff</enum>
                </property>
                <property name="horizontalScrollBarPolicy">
                 <enum>Qt::ScrollBarAlwaysOff</enum>
                </property>
                <property name="sizeAdjustPolicy">
                 <enum>QAbstractScrollArea::AdjustToContents</enum>
                </property>
                <column>
                 <property name="text">
                  <string>Assignment</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Grade</string>
                 </property>
                </column>
                <column>
                 <property name="text">
                  <string>Letter Grade</string>
                 </property>
                </column>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="Profile">
       <attribute name="title">
        <string>  Profile  </string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QFrame" name="sunnydale_crest_frame">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QFrame" name="crest_frame">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_8">
              <item>
               <widget class="QLabel" name="sunnydale_logo_png">
                <property name="maximumSize">
                 <size>
                  <width>500</width>
                  <height>500</height>
                 </size>
                </property>
                <property name="text">
                 <string>Sunnydale Logo</string>
                </property>
                <property name="scaledContents">
                 <bool>true</bool>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QFrame" name="information_frame">
             <property name="frameShape">
              <enum>QFrame::StyledPanel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Raised</enum>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_12">
              <item>
               <widget class="QGroupBox" name="groupBox">
                <property name="title">
                 <string/>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_4">
                 <item>
                  <widget class="QLabel" name="labelTeacherInfo_3">
                   <property name="font">
                    <font>
                     <family>Segoe UI,sans-serif</family>
                     <pointsize>16</pointsize>
                     <weight>75</weight>
                     <bold>true</bold>
                    </font>
                   </property>
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 16pt;
   	font-weight: bold;
</string>
                   </property>
                   <property name="text">
                    <string>Homeroom Teacher Info</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="labelTeacherName_3">
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 13pt;
</string>
                   </property>
                   <property name="text">
                    <string>Teacher Name: [DYNAMIC: Teacher Name]</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="labelTeacherEmail_3">
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 13pt;
</string>
                   </property>
                   <property name="text">
                    <string>Email: [DYNAMIC: Teacher Email]</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBoxStudentInfo">
                <property name="title">
                 <string/>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_6">
                 <item>
                  <widget class="QLabel" name="labelStudentInfo_3">
                   <property name="font">
                    <font>
                     <family>Segoe UI,sans-serif</family>
                     <pointsize>16</pointsize>
                     <weight>75</weight>
                     <bold>true</bold>
                    </font>
                   </property>
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 16pt;
   	font-weight: bold;
</string>
                   </property>
                   <property name="text">
                    <string>Student Info</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="labelStudentName_5">
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 13pt;
</string>
                   </property>
                   <property name="text">
                    <string>[DYNAMIC: Student First and Last Name]</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="labelStudentDOB_3">
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 13pt;
</string>
                   </property>
                   <property name="text">
                    <string>[DYNAMIC: Student DOB]</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="labelStudentGradeLevel_3">
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 13pt;
</string>
                   </property>
                   <property name="text">
                    <string>[DYNAMIC: Student Grade Level]</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
              <item>
               <widget class="QGroupBox" name="groupBoxGuardianInfo">
                <property name="title">
                 <string/>
                </property>
                <layout class="QVBoxLayout" name="verticalLayout_5">
                 <item>
                  <widget class="QLabel" name="labelGuardianInfo_3">
                   <property name="font">
                    <font>
                     <family>Segoe UI,sans-serif</family>
                     <pointsize>16</pointsize>
                     <weight>75</weight>
                     <bold>true</bold>
                    </font>
                   </property>
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 16pt;
   	font-weight: bold;
</string>
                   </property>
                   <property name="text">
                    <string>Guardian Info</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="labelGuardianName_3">
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 13pt;
</string>
                   </property>
                   <property name="text">
                    <string>[DYNAMIC: Guardian name]</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QLabel" name="labelGuardianPhone_3">
                   <property name="styleSheet">
                    <string notr="true">
	qproperty-alignment: 'AlignCenter';
   	font-size: 13pt;
</string>
                   </property>
                   <property name="text">
                    <string>[DYNAMIC: Guardian Phone Number]</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1100</width>
     <height>0</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
